tests/mpi/project_bv_div_conf.cc

```diff
@@ -0,0 +1,290 @@
+						 p2, colorize);
+      triangulation.refine_global(2);
+    } // end make_grid
+
+  template<int dim>
+    void
+    ResFlowProblem<dim>::setup_system ()
+    {
+
+      dof_handler.distribute_dofs (fe);
+      std::vector<unsigned int> resflow_sub_blocks (2, 0);
+      resflow_sub_blocks[1] = 1;
+      // Renumber to yield block structure
+      DoFRenumbering::block_wise (dof_handler);
+
+      std::vector<types::global_dof_index> dofs_per_block (2);
+      DoFTools::count_dofs_per_block (dof_handler, dofs_per_block,
+				      resflow_sub_blocks);
+
+      const unsigned int n_u = dofs_per_block[0], n_p = dofs_per_block[1];
+      pcout << "   Number of degrees of freedom: " << dof_handler.n_dofs ()
+	  << " (" << n_u << '+' << n_p << ')' << std::endl;
+
+      // We split up the IndexSet for locally owned and locally relevant DoFs
+      // into two IndexSets based on how we want to create the block matrices
+      // and vectors.
+      owned_partitioning.resize (2);
+      owned_partitioning[0] = dof_handler.locally_owned_dofs ().get_view (0,
+									  n_u);
+      owned_partitioning[1] = dof_handler.locally_owned_dofs ().get_view (
+	  n_u, n_u + n_p);
+
+      pcout << "Number of Owned dofs: " << dof_handler.n_locally_owned_dofs ()
+	  << " (" << owned_partitioning[0].n_elements () << '+'
+	  << owned_partitioning[1].n_elements () << ')' << std::endl
+	  << std::endl;
+
+      IndexSet locally_relevant_dofs;
+      DoFTools::extract_locally_relevant_dofs (dof_handler,
+					       locally_relevant_dofs);
+      relevant_partitioning.resize (2);
+      relevant_partitioning[0] = locally_relevant_dofs.get_view (0, n_u);
+      relevant_partitioning[1] = locally_relevant_dofs.get_view (n_u,
+								 n_u + n_p);
+
+      pcout << "Number of Relevant dofs: "
+	  << locally_relevant_dofs.n_elements () << " ("
+	  << relevant_partitioning[0].n_elements () << '+'
+	  << relevant_partitioning[1].n_elements () << ')' << std::endl
+	  << std::endl;
+
+	{ 
+	  constraints.reinit (locally_relevant_dofs);
+
+	  FEValuesExtractors::Vector velocities (0);
+	  DoFTools::make_hanging_node_constraints (dof_handler, constraints);
+
+	  VectorTools::project_boundary_values_div_conforming (
+	      dof_handler,
+	      0,
+	      FluxBoundaryValues<dim> (),
+	      0,
+	      constraints);
+	  if ((Utilities::MPI::this_mpi_process (mpi_communicator)) == 0)
+	      constraints.print(std::cout);
+	  constraints.close ();
+	}
+    }	// end setup_system
+
+
+  template<int dim>
+    void
+    ResFlowProblem<dim>::run ()
+    {
+      pcout << "Running with "
+	  << Utilities::MPI::n_mpi_processes (mpi_communicator)
+	  << " MPI rank(s)..." << std::endl;
+
+      make_grid ();
+      pcout << "Grid made." << std::endl;
+      setup_system ();
+      pcout << "System set up." << std::endl;
+    }
+
+} // end namespace ResFlow
+
+int
+main (int argc, char *argv[])
+{
+  const unsigned int dim = 2;
```

Comment:
You'll have to add some lines like [this](https://github.com/dealii/dealii/blob/master/tests/mpi/attach_data_02.cc#L202-L203) here:
```c++
Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
MPILogInitAll log;
```